### PR TITLE
.Net: Remove dependency on planners.core

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

New planners must not depend on planners core so they can be taken to v1.0 independently

### Description

Move the shared code to internal utilities

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/SemanticKernel.Core/Planning/PlannerConfigBase.cs
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel.Core/Planning/PlannerConfigBase.cs:1]: Shouldn't this file keep in Planners.Core ?

File: dotnet/src/SemanticKernel.Core/Planning/PlannerInstrumentation.cs
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel.Core/Planning/PlannerInstrumentation.cs:1]: Same for this?

**Review by RogerBarreto — APPROVED** [review]
LGTM, Not sure if we should keep any Plan file in Kernel.Core.

File: dotnet/src/SemanticKernel.Core/Planning/PlannerConfigBase.cs
Comment (markwallace-microsoft) [reply dotnet/src/SemanticKernel.Core/Planning/PlannerConfigBase.cs:1]: Moving to the internal utilities as it's shared with the openai planner

File: dotnet/src/SemanticKernel.Core/Planning/PlannerInstrumentation.cs
Comment (markwallace-microsoft) [reply dotnet/src/SemanticKernel.Core/Planning/PlannerInstrumentation.cs:1]: Moving to the internal utilities as it's shared with the openai planner

**Review by SergeyMenshykh — APPROVED** [review]

**Pull request merged**